src/libstore/include/nix/store/store-dir-config.hh

```diff
@@ -18,24 +18,32 @@ struct SourcePath;
 MakeError(BadStorePath, Error);
 MakeError(BadStorePathName, BadStorePath);
 
-struct StoreDirConfig : public Config
+/**
+ * Underlying store directory configuration type.
+ *
+ * Don't worry to much about the `F` parameter, it just some abstract
+ * nonsense for the "higher-kinded data" pattern. It is used in each
+ * settings record in order to ensure don't forgot to parse or document
+ * settings field.
+ */
+template<template<typename> class F>
+struct StoreDirConfigT
 {
-    using Config::Config;
-
-    StoreDirConfig() = delete;
-
-    virtual ~StoreDirConfig() = default;
-
-    const PathSetting storeDir_{this, settings.nixStore,
-        "store",
-        R"(
-          Logical location of the Nix store, usually
-          `/nix/store`. Note that you can only copy store paths
-          between stores if they have the same `store` setting.
-        )"};
-    const Path storeDir = storeDir_;
+    F<Path> _storeDir;
```

Comment:
If `use Setting<F, Path> = F<Path>;` works and needs no extra initializer braces:

```suggestion
    Setting<F, Path> _storeDir;
```